src/Resources/Resources.Test/ScenarioTests/DeploymentTests.ps1

```diff
@@ -884,4 +904,62 @@ function Test-TestDeploymentFromBicepFile
         # Cleanup
         Clean-ResourceGroup $rgname
     }
+}
+
+<#
+.SYNOPSIS
+Tests what-if on a deployment from a template in a storage account using a query string.
+#>
+function Test-WhatIfWithQueryString
+{
+	# Setup
+    $rgname = Get-ResourceGroupName
+    $rname = Get-ResourceName
+	$saname = "querystringpstests"
+    $rglocation = "West US 2"
+    $subId = (Get-AzContext).Subscription.SubscriptionId
+
+	try
+	{
+		# Prepare our RG
+        New-AzResourceGroup -Name $rgname -Location $rglocation
+
+		#Prepare our Storage Account
+		$account = New-AzStorageAccount -ResourceGroupName $rgname -AccountName $saname -Location $rglocation -SkuName "Standard_LRS"
+
+		#Get StorageAccountKey
+		$key = (Get-AzStorageAccountKey -ResourceGroupName $rgname -AccountName $saname)| Where-Object {$_.KeyName -eq "key1"}
+
+		#Get StorageAccount context
+		$context = New-AzStorageContext -StorageAccountName $saname -StorageAccountKey $key.Value
+
+		#Create FileShare
+		New-AzStorageShare -Name "querystringshare" -Context $context
+
+		#Upload files to the StorageAccount
+		Set-AzStorageFileContent -ShareName "querystringshare" -Source "sampleLinkedTemplateParent.json" -Path "sampleLinkedTemplateParent.json" -Context $context
+		Set-AzStorageFileContent -ShareName "querystringshare" -Source "sampleLinkedTemplateChild.json" -Path "sampleLinkedTemplateChild.json" -Context $context
+
+		#Get SAStoken
+		$token = New-AzStorageAccountSASToken -Service File -ResourceType Service,Container,Object -Permission "r" -Context $context -ExpiryTime (Get-Date).AddMinutes(3)
+
+		#Create deployment
+		$deployment =New-AzResourceGroupDeployment -Name $rname -ResourceGroupName $rgname -TemplateUri "https://querystringpstests.file.core.windows.net/querystringshare/sampleLinkedTemplateParent.json" -QueryString $token.Substring(1)
```

Comment:
Is this URL valid? I cannot access it in my environment.

Reply:
Same reasoning as the comment below